hazelcast/include/hazelcast/client/connection/ClientConnectionManagerImpl.h

```diff
@@ -188,7 +188,11 @@ class HAZELCAST_API ClientConnectionManagerImpl
         if (member_addresses.empty()) {
             return;
         }
-        std::random_shuffle(member_addresses.begin(), member_addresses.end());
+
+        std::random_device rd;
+        std::mt19937 gen(rd());
```

Comment:
check [this](https://stackoverflow.com/questions/72577246/why-is-random-device-creation-expensive) link on the cost of these generator constructions. We can cache it to decrease the cost.

Reply:
Fixed.

Btw, mean of creating `std::random_device` and `std::mt19937` is about `2us` on my computer but doesn't matter I cached it.